### PR TITLE
Implements --version flag

### LINKED ISSUE #3239: Add a `--version` flag
### What problem are you facing?

I'm currently trying to figure out what version of Crossplane a particular build is by creating a `Configuration` that will only work on that version, e.g.

```yaml
apiVersion: meta.pkg.crossplane.io/v1alpha1
kind: Configuration
metadata:
  name: is-this-the-right-version
spec:
  crossplane:
    version: "v1.9.0-up.2"
```

### How could Crossplane help solve your problem?

Life would be a lot simpler if I could just run `crossplane --version` to find out what version a particular binary was.

### PR DESCRIPTION
**Pull request by Santosh1176**
Signed-off-by: Santosh Kaluskar <dtshbl@gmail.com>

### Description of your changes
Implements `--version` flag.

Fixes #3239

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested
I could not build the local dev env. Hence, I could not test the command. Apologies for any inconvenience caused by that.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by negz** [comment]
> I could not build the local dev env. Hence, I could not test the command. Apologies for any inconvenience caused by that.

We'll definitely need to run this to make sure it works before we can merge it. Does https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md#establishing-a-development-environment help? Where are you getting stuck?

File: cmd/crossplane/main.go
Comment (jbw976) [inline cmd/crossplane/main.go:42]: declaring versionFlag as a string is (AFAICT) telling Kong to expect a string value to be passed in along with the `-v`.   Note the `-v, --version=VERSION-FLAG` in the help output.

Just running `crossplane -v` results in the help output and an error message of `crossplane: error: --version: expected string value but got "EOL" (<EOL>)`.

Passing a dummy value for `-v` results in correct operation (the crossplane binary version is printed), giving further evidence that is expecting a string to be passed as the option's value:
```
❯ _output/bin/darwin_arm64/crossplane --v=foo
v1.10.0-rc.0.92.ge7c31b08
```

Instead of a string, this should be a flag/switch - see if you can research how Kong treats those types of command line options 🙏 

full output:
```
/dev/go/src/github.com/crossplane/crossplane @e7c31b08
❯ _output/bin/darwin_arm64/crossplane -v
Usage: crossplane <command>

An open source multicloud control plane.

Flags:
  -h, --help                    Show context-sensitive help.
  -d, --debug                   Print verbose logging statements.
  -v, --version=VERSION-FLAG    Print version and quit.

Commands:
  core start
    Start Crossplane controllers.

  core init
    Make cluster ready for Crossplane controllers.

  rbac start
    Start Crossplane RBAC controllers.

  rbac init
    Initialize RBAC Manager.

Run "crossplane <command> --help" for more information on a command.

crossplane: error: --version: expected string value but got "EOL" (<EOL>)
```

**Review by jbw976 — COMMENTED** [review]
Thanks for taking a stab at this @Santosh1176! Hopefully we can get your development environment working OK so that you can test your changes and iterate over them to get them working successfully 🙏

**Comment by jbw976** [comment]
From https://github.com/crossplane/crossplane/issues/3239#issuecomment-1222907898, looks like @Santosh1176 is getting stuck at the `make` output of `verify dependencies have expected content`.  The command being run during this appears to be `go mod verify` as per https://github.com/upbound/build/blob/559e1532cd1ec6471c295b9dfce89c38f9a4638f/makelib/golang.mk#L217-L218.

@Santosh1176 can you run `go mod verify` directly (without make) successfully?  What version of go do you have installed?

**Comment by Santosh1176** [comment]
@jbw976 Thank you for helping me out in building the local development environment and pointing out the error in my implementation. I shall work on this and get this resolved. Regarding the local environment, I retried to again, and following are the results:
`go mod verify` returns "all modules verified". Further, I was successfully able to establish the build submodule with `_output` directory. But, from `_output/bin/linux_amd64` when I try to run `crossplane --help` or `crossplane --version` I get "crossplane: command not found". How do I resolve this and move ahead?
Screenshot from 2022-08-24 22-18-54

**Comment by jbw976** [comment]
Hmm - so `make` from the root of the crossplane repo also completes successfully for you now too? I wanted you to run `go mod verify` directly, just to see how it behaved outside of its invocation from within the `make` flow.

But `make` works OK for you now too? what are the contents of that `_output/bin/linux_amd64` directory, can you do a `ls -al` on it to see the full contents? 🤔

**Comment by Santosh1176** [comment]
> Hmm - so `make` from the root of the crossplane repo also completes successfully for you now too? I wanted you to run `go mod verify` directly, just to see how it behaved outside of its invocation from within the `make` flow.
> 
> But `make` works OK for you now too? what are the contents of that `_output/bin/linux_amd64` directory, can you do a `ls -al` on it to see the full contents? thinking

Just realized, I need to run `make` after any changes are made to the code. Now the local build works as intended. Thank you for your support and patience. 

I made some changes to the implementation of the version flag and I am getting the following output:

```bash
❯ crossplane/_output/bin/linux_amd64/crossplane --version
v0.0.0-4043.ge7c31b08.dirty
```
Is this the desired outcome?

crossplane-flag

**Comment by jbw976** [comment]
That does look like the correct behavior of the `-v` and `--version` flags @Santosh1176! i'm curious why your printed version is `v0.0.0*` instead of what I saw that started with `v1.10.0*` 🤔 looks like this could have to do with the existence of tags or not from https://github.com/upbound/build/blob/559e1532cd1ec6471c295b9dfce89c38f9a4638f/makelib/common.mk#L217-L223.  

Could you push your latest code so we can review the latest changes you've made?

File: cmd/crossplane/main.go
Comment (jbw976) [inline cmd/crossplane/main.go:47]: small nit: having the `short` field listed first, similar to the `debugFlag`, may make it easier and more consistent to read.

**Review by jbw976 — APPROVED** [review]
great @Santosh1176! i tested this latest version out locally and the functionality looks good to me.  Looks like it follows similar patterns that I see in https://github.com/upbound/up/blob/main/cmd/up/main.go#L50-L54, so I feel good about that too.

LGTM, but i'll hold off from merging for an official maintainer to do so 😓 

/cc @negz @muvaf @hasheddan @turkenh

File: cmd/crossplane/main.go
Comment (Santosh1176) [reply cmd/crossplane/main.go]: Done!

**Review by negz — APPROVED** [review]
Thanks @Santosh1176!

**Comment by Santosh1176** [comment]
> Thanks @Santosh1176!

Thank you!

**Pull request merged**
